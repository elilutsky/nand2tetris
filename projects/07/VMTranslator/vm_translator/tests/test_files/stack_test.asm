@17
D=A
@SP
A=M
M=D
@SP
M=M+1
@17
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@stack_test$_comparsion_true.0
D;JEQ
D=0
@stack_test$_comparsion_end.0
0;JMP
(stack_test$_comparsion_true.0)
D=-1
(stack_test$_comparsion_end.0)
@SP
A=M
M=D
@SP
M=M+1
@17
D=A
@SP
A=M
M=D
@SP
M=M+1
@16
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@stack_test$_comparsion_true.1
D;JEQ
D=0
@stack_test$_comparsion_end.1
0;JMP
(stack_test$_comparsion_true.1)
D=-1
(stack_test$_comparsion_end.1)
@SP
A=M
M=D
@SP
M=M+1
@16
D=A
@SP
A=M
M=D
@SP
M=M+1
@17
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@stack_test$_comparsion_true.2
D;JEQ
D=0
@stack_test$_comparsion_end.2
0;JMP
(stack_test$_comparsion_true.2)
D=-1
(stack_test$_comparsion_end.2)
@SP
A=M
M=D
@SP
M=M+1
@892
D=A
@SP
A=M
M=D
@SP
M=M+1
@891
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@stack_test$_comparsion_true.3
D;JLT
D=0
@stack_test$_comparsion_end.3
0;JMP
(stack_test$_comparsion_true.3)
D=-1
(stack_test$_comparsion_end.3)
@SP
A=M
M=D
@SP
M=M+1
@891
D=A
@SP
A=M
M=D
@SP
M=M+1
@892
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@stack_test$_comparsion_true.4
D;JLT
D=0
@stack_test$_comparsion_end.4
0;JMP
(stack_test$_comparsion_true.4)
D=-1
(stack_test$_comparsion_end.4)
@SP
A=M
M=D
@SP
M=M+1
@891
D=A
@SP
A=M
M=D
@SP
M=M+1
@891
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@stack_test$_comparsion_true.5
D;JLT
D=0
@stack_test$_comparsion_end.5
0;JMP
(stack_test$_comparsion_true.5)
D=-1
(stack_test$_comparsion_end.5)
@SP
A=M
M=D
@SP
M=M+1
@32767
D=A
@SP
A=M
M=D
@SP
M=M+1
@32766
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@stack_test$_comparsion_true.6
D;JGT
D=0
@stack_test$_comparsion_end.6
0;JMP
(stack_test$_comparsion_true.6)
D=-1
(stack_test$_comparsion_end.6)
@SP
A=M
M=D
@SP
M=M+1
@32766
D=A
@SP
A=M
M=D
@SP
M=M+1
@32767
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@stack_test$_comparsion_true.7
D;JGT
D=0
@stack_test$_comparsion_end.7
0;JMP
(stack_test$_comparsion_true.7)
D=-1
(stack_test$_comparsion_end.7)
@SP
A=M
M=D
@SP
M=M+1
@32766
D=A
@SP
A=M
M=D
@SP
M=M+1
@32766
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@stack_test$_comparsion_true.8
D;JGT
D=0
@stack_test$_comparsion_end.8
0;JMP
(stack_test$_comparsion_true.8)
D=-1
(stack_test$_comparsion_end.8)
@SP
A=M
M=D
@SP
M=M+1
@57
D=A
@SP
A=M
M=D
@SP
M=M+1
@31
D=A
@SP
A=M
M=D
@SP
M=M+1
@53
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=D+A
@SP
A=M
M=D
@SP
M=M+1
@112
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
D=-D
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=D&A
@SP
A=M
M=D
@SP
M=M+1
@82
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=D|A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
D=!D
@SP
A=M
M=D
@SP
M=M+1
