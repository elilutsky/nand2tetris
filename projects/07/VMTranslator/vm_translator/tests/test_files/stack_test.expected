@17
D=A
@SP
A=M
M=D
@SP
M=M+1
@17
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@_LOGICAL_SET_TRUE_2
D;JEQ
D=0
@_LOGICAL_END_2
0;JMP
(_LOGICAL_SET_TRUE_2)
D=-1
(_LOGICAL_END_2)
@SP
A=M
M=D
@SP
M=M+1
@17
D=A
@SP
A=M
M=D
@SP
M=M+1
@16
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@_LOGICAL_SET_TRUE_5
D;JEQ
D=0
@_LOGICAL_END_5
0;JMP
(_LOGICAL_SET_TRUE_5)
D=-1
(_LOGICAL_END_5)
@SP
A=M
M=D
@SP
M=M+1
@16
D=A
@SP
A=M
M=D
@SP
M=M+1
@17
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@_LOGICAL_SET_TRUE_8
D;JEQ
D=0
@_LOGICAL_END_8
0;JMP
(_LOGICAL_SET_TRUE_8)
D=-1
(_LOGICAL_END_8)
@SP
A=M
M=D
@SP
M=M+1
@892
D=A
@SP
A=M
M=D
@SP
M=M+1
@891
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@_LOGICAL_SET_TRUE_11
D;JLT
D=0
@_LOGICAL_END_11
0;JMP
(_LOGICAL_SET_TRUE_11)
D=-1
(_LOGICAL_END_11)
@SP
A=M
M=D
@SP
M=M+1
@891
D=A
@SP
A=M
M=D
@SP
M=M+1
@892
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@_LOGICAL_SET_TRUE_14
D;JLT
D=0
@_LOGICAL_END_14
0;JMP
(_LOGICAL_SET_TRUE_14)
D=-1
(_LOGICAL_END_14)
@SP
A=M
M=D
@SP
M=M+1
@891
D=A
@SP
A=M
M=D
@SP
M=M+1
@891
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@_LOGICAL_SET_TRUE_17
D;JLT
D=0
@_LOGICAL_END_17
0;JMP
(_LOGICAL_SET_TRUE_17)
D=-1
(_LOGICAL_END_17)
@SP
A=M
M=D
@SP
M=M+1
@32767
D=A
@SP
A=M
M=D
@SP
M=M+1
@32766
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@_LOGICAL_SET_TRUE_20
D;JGT
D=0
@_LOGICAL_END_20
0;JMP
(_LOGICAL_SET_TRUE_20)
D=-1
(_LOGICAL_END_20)
@SP
A=M
M=D
@SP
M=M+1
@32766
D=A
@SP
A=M
M=D
@SP
M=M+1
@32767
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@_LOGICAL_SET_TRUE_23
D;JGT
D=0
@_LOGICAL_END_23
0;JMP
(_LOGICAL_SET_TRUE_23)
D=-1
(_LOGICAL_END_23)
@SP
A=M
M=D
@SP
M=M+1
@32766
D=A
@SP
A=M
M=D
@SP
M=M+1
@32766
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@_LOGICAL_SET_TRUE_26
D;JGT
D=0
@_LOGICAL_END_26
0;JMP
(_LOGICAL_SET_TRUE_26)
D=-1
(_LOGICAL_END_26)
@SP
A=M
M=D
@SP
M=M+1
@57
D=A
@SP
A=M
M=D
@SP
M=M+1
@31
D=A
@SP
A=M
M=D
@SP
M=M+1
@53
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=D+A
@SP
A=M
M=D
@SP
M=M+1
@112
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=A-D
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
D=-D
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=D&A
@SP
A=M
M=D
@SP
M=M+1
@82
D=A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
@SP
M=M-1
@SP
A=M
A=M
D=D|A
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
@SP
A=M
D=M
D=!D
@SP
A=M
M=D
@SP
M=M+1